src/dotnet/CarbonAware.WebApi/Controllers/SciScoreController.cs

```diff
@@ -1,45 +1,125 @@
 
-    public SciScoreController(ILogger<SciScoreController> logger)
+    public SciScoreController(ILogger<SciScoreController> logger, ISciScoreAggregator aggregator)
     {
         _logger = logger;
+        _aggregator = aggregator;
     }
 
     [HttpPost]
     [Consumes(MediaTypeNames.Application.Json)]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
-    public async Task<IActionResult> CreateAsync(SciScoreCalculation calculation)
+
+    public async Task<IActionResult> CreateAsync(SciScoreInput input)
     {
-        if (String.IsNullOrEmpty(calculation.AzRegion))
+        if (input.Location == null)
         {
-            return BadRequest("AzRegion is required");
+            var error = new CarbonAwareWebApiError() { Message = "Location is required" };
+            return BadRequest(error);
         }
 
-        if (String.IsNullOrEmpty(calculation.Duration))
+        if (String.IsNullOrEmpty(input.TimeInterval))
         {
-            return BadRequest("Duration is required");
+            var error = new CarbonAwareWebApiError() { Message = "TimeInterval is required" };
+            return BadRequest(error);
         }
 
         SciScore score = new SciScore
+        {
+            SciScoreValue = 100.0,
+            EnergyValue = 1.0,
+            MarginalCarbonEmissionsValue = 100.0,
+            EmbodiedEmissionsValue = 0.0,
+            FunctionalUnitValue = 1
+        };
+
+        return await Task.Run(() => Ok(score));
+    }
+
+    [HttpPost("marginal-carbon-intensity")]
+    [Consumes(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<IActionResult> GetCarbonIntensityAsync(SciScoreInput input)
+    {
+        _logger.LogInformation(" calling to aggregator to ");
+        if (input.Location == null)
+        {
+            var error = new CarbonAwareWebApiError() { Message = "location is required" };
+            return BadRequest(error);
+        }
+
+        if (String.IsNullOrEmpty(input.TimeInterval))
+        {
+            var error = new CarbonAwareWebApiError() { Message = "timeInterval is required" };
+            return BadRequest(error);
+        }
+
+        try
+        {
+            var location = this.GetLocation(input.Location);
+            var carbonIntensity = await _aggregator.CalculateAverageCarbonIntensityAsync(location, input.TimeInterval);
+            SciScore score = new SciScore
             {
-                SciScoreValue = 100.0f,
-                EnergyValue = 1.0f,
-                MarginalCarbonEmissionsValue = 100.0f,
-                EmbodiedEmissionsValue = 0.0f,
-                FunctionalUnitValue = 1
+                MarginalCarbonEmissionsValue = carbonIntensity,
             };
 
-        return await Task.Run(() => Ok(score));
+            return Ok(score);
+        }
+        catch (Exception ex)
```

Comment:
:sheep: maybe make a note here of what potential exceptions we expect (location and/or calculating average carbon intensity)

Reply:
Noted in a comment. Though I think when we address the WattTimeException story we may want to scope this down to just  expected error classes and fail more loudly for unhandled errors.